core/tee/tee_svc_cryp.c

```diff
@@ -1017,10 +1018,10 @@ TEE_Result tee_obj_attr_to_binary(struct tee_obj *o, void *data,
 		const struct tee_cryp_obj_type_attrs *ta = tp->type_attrs + n;
 		void *attr = (uint8_t *)o->attr + ta->raw_offs;
 
-		attr_ops[ta->ops_index].to_binary(attr, data, *data_len, &offs);
+		attr_ops[ta->ops_index].to_binary(attr, data, len, &offs);
 	}
 
-	if (offs > *data_len)
+	if (data && offs > *data_len)
```

Comment:
With this change all places that call this function to query required size shouldn't test for `TEE_ERROR_SHORT_BUFFER` as return value as it shouldn't happen any longer.